sale_partner_insurance/models/sale_order.py

```diff
@@ -0,0 +1,82 @@
+# Copyright 2024 Akretion France (http://www.akretion.com/)
+# @author: Mathieu Delva <mathieu.delva@akretion.com>
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+from odoo import fields, models
+
+
+class SaleOrder(models.Model):
+    _inherit = "sale.order"
+
+    def action_confirm(self):
+        for record in self:
+            partner_insurance = record.partner_id.coefficient_sale_insurance
```

Comment:
Better to check on commercial_partner_id if the partner of the sale order is a contact for example. 
```suggestion
            partner_insurance = record.commercial_partner_id.coefficient_sale_insurance
```